src/utils/i18n-utils.js

```diff
@@ -25,16 +25,34 @@ export function localeIsValid(localeCode) {
 
 /**
  * Updates i18n global var and html lang
- * @param {{ params: { locale: String } }} to - where the route navigates to
+ * @param {String} locale - locale used
  * @param {Object{}} env - context
- * @param {{ code: String, name: String }[]} locales
  */
-export function updateCurrentLocale(to, env) {
-  const currentLocale = to.params.locale || defaultLocale;
+export function updateLocale(locale = defaultLocale, env) {
   // exist if current locale is not supported
-  if (!localeIsValid(currentLocale)) return;
+  if (!localeIsValid(locale)) return;
   // update locale global var
-  env.$i18n.locale = currentLocale; // eslint-disable-line no-param-reassign
+  env.$i18n.locale = locale; // eslint-disable-line no-param-reassign
   // update html lang
-  updateLangTag(currentLocale);
+  updateLangTag(locale);
+}
+
+/**
+ * Updates i18n global var and html lang
+ * @param {{ params: { locale: String } }} to - where the route navigates to
+ * @param {Object{}} env - context
+ */
+export function updateCurrentLocale(to, env) {
+  const currentLocale = to.params.locale;
+  updateLocale(currentLocale, env);
```

Comment:
This is missing code to map the URL slug (`currentLocale`) to the appropriate locale code that `$i18n` expects and only works here, because it assumes that the slug will always be an exact match.